src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/ProjectSystem/RazorProjectService.TestAccessor.cs

```diff
@@ -13,5 +18,24 @@ internal readonly struct TestAccessor(RazorProjectService instance)
     {
         public ValueTask WaitForInitializationAsync()
             => instance.WaitForInitializationAsync();
+
+        public async Task<ProjectKey> AddProjectAsync(
+            string filePath,
+            string intermediateOutputPath,
+            RazorConfiguration? configuration,
+            string? rootNamespace,
+            string? displayName,
+            CancellationToken cancellationToken)
+        {
+            var service = instance;
```

Comment:
❓ why?

Reply:
Short answer: it makes the compiler happy

Long answer: The test accessor is a struct, and a lambda in a struct can't access a member of the struct. Or at least that's what I discovered when the compiler complained at me :)

Reply:
ah interesting, so this is causing it to box? Will have to look more, but at first glance it was weird. Newfangled C#

Reply:
Yes, it would have to box the instance, and I guess they want that to be explicit and not hidden. I don't begin to understand this low level stuff :)

The error was CS1673 IIRC and you wanted to look up more about it.

Reply:
Yes, this is a problem because you're capturing a primary constructor parameter that is ultimately backed by a field on the struct. So, you're _really_ capturing a field which means that the lambda must capture `this`, which requires boxing the struct. Otherwise, the closure will contain a copy of the struct.

Reply:
Here are the two errors @davidwengier likely ran into:

https://sharplab.io/#v2:EYLgtghglgdgNAExAagD4AEBMBGAsAKHQAYACdbAOgBlYBHAbgIPQGYSAnAUwgQHsYANgE8SAZwAu7AK4BjcSQDK2ABTlSogJQkA3gRL6ybYL14CSAYX7joMUauxEA2gF0SEduwhCtu/Af9kAOxuHl4UAOoAFpxcygAeJAC8AHwkCYmJYhoUAIIwQsoajH4GAL4E5fjMbFw8/MJikrLyCpj26j56BgAO7FAAbhDinBzcfIIiaiQA+qJJYsX+XfqsJMamFlY2dmouIZ7eOssB6MHuBxHRsemp6Zmz2XkFRceVpUA=

Reply:
Yea, this definitely makes sense. In a PR it looks weird but wholistically in the language is :+1: